[waihin26] iP

waihin26 opened the pull request:
# My Awesome Project

> “Simplicity is the ultimate sophistication.” – Leonardo da Vinci

This project helps you simplify your life by managing your tasks with ease. It’s:

- **User-friendly** 😄
- **Efficient** 😎
- **SUPER FAST** 💨

### Getting Started

To start using the project, follow these simple steps:

1. **Download** the project from [here](https://github.com/waihin26/ip).
2. **Install** it by following the installation instructions.
3. **Run** the project to start managing your tasks.
4. **Enjoy** the simplicity it brings to your life 😊.

And the best part? It's **FREE**!

### Features:

- [x] Task Management
- [ ] Deadline Management (coming soon)
- [ ] Reminder Notifications (coming soon)

### Example Code:

```java
 public static void main(String[] args) {
        new Blue().run();
 }

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

waihin26 committed:
Initialised skeleton of chatbot Blue

waihin26 committed:
Implemented scanner such that chatbot echos user

waihin26 committed:
Added dog image

waihin26 committed:
Created Note.java file

waihin26 committed:
Added list feature to Blue.java file

waihin26 committed:
Created Task.java file

waihin26 committed:
Cleaned up Note.java and Task.java codes

waihin26 committed:
Added mark and unmark functions to chatbot

waihin26 committed:
Created Todo.java class

waihin26 committed:
Created Deadline.java class

waihin26 committed:
Created Event.java class

waihin26 committed:
Updated Blue.java and Note.java codes to support the 3 different tasks

waihin26 committed:
Touch up to files to ensure consistency with level 4 product

waihin26 committed:
Created exception classes to handle empty descriptions and unkwown inputs

waihin26 committed:
Delete Function added

waihin26 committed:
Touch up of codes

waihin26 committed:
Added automatic testing feature

waihin26 committed:
further cleaning up of codes

waihin26 committed:
Fixed string description of Deadline event

waihin26 committed:
add suffix to Deadline, Event and ToDo class

waihin26 committed:
add abstract toFileString method

waihin26 committed:
add fromFileString method

waihin26 committed:
add method to write into txt file

waihin26 committed:
add WrongNumberOfItemException

waihin26 committed:
Note reads from file and writes to file whenever tasks list updated

waihin26 committed:
Merge branch 'branch-Level-7'

waihin26 committed:
understand date and time inputs for deadline tasks

waihin26 committed:
Merge branch 'branch-Level-8'

waihin26 committed:
edit bug related to datetime formatter

waihin26 committed:
create UI and Parser class.

waihin26 committed:
Update storage class to handle the loading and saving to files

waihin26 committed:
Add Parser class and UI class to clean up code with OOP concepts

waihin26 committed:
Merge branch 'OOP'

waihin26 committed:
refactor codes into different packages

waihin26 committed:
Merge branch 'add-gradle-support'

waihin26 committed:
set up A-Gradle

waihin26 committed:
add J-unit testing for fromFileString and toFileString method

waihin26 committed:
edit build.gradle

waihin26 committed:
add JavaDocs comments to all files

waihin26 committed:
follow coding conventions

waihin26 committed:
Merge branch 'branch-A-JavaDoc'

waihin26 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/blue/UI.java
#	src/main/java/blue/task/Task.java

waihin26 committed:
edit style

waihin26 committed:
add find method to chatbot

waihin26 committed:
Merge branch 'branch-Level-9'

waihin26 committed:
conform to coding standards

waihin26 committed:
Merge branch 'A-CheckStyle'

yxshrk commented on src/main/java/blue/Blue.java:
```diff
@@ -0,0 +1,41 @@
+ */
+public class Blue {
+    /** The task list managed by the application. */
+    private TaskList tasklist;
```
I noticed that you named the variable tasklist here. According to naming conventions, it’s better to use taskList with camel case. Would it be clearer to rename this as taskList for consistency with standard Java conventions?

yxshrk commented on src/main/java/blue/Blue.java:
```diff
@@ -0,0 +1,41 @@
+    private TaskList tasklist;
+
+    /** The parser that handles user input. */
+    private Parser parser;
```
The name parser is well-chosen and follows Java naming conventions. I like how the variable name is clear and indicates its purpose.

yxshrk commented on src/main/java/blue/UI.java:
```diff
@@ -0,0 +1,96 @@
+     *
+     * @param task The task that has been marked as done.
+     */
+    public static void displayAfterMark(Task task) {
```
Would it be clearer to rename displayAfterMark to displayTaskMarkedAsDone? This would more explicitly describe what the method is doing, aligning better with standard naming conventions that prioritize clarity.

yxshrk commented on src/main/java/blue/UI.java:
```diff
@@ -0,0 +1,96 @@
+     *
+     * @param task The task that has been unmarked.
+     */
+    public static void displayAfterUnMark(Task task) {
```
Similarly, how about renaming displayAfterUnMark to displayTaskMarkedAsNotDone? This change could improve the clarity and make it consistent with naming conventions that emphasize descriptive method names.

yxshrk reviewed (commented):
Looks good to me, just some adjustments needed to follow naming conventions.

ruiming97 commented on src/main/java/blue/UI.java:
```diff
@@ -0,0 +1,96 @@
+    /**
+     * Displays a farewell message to the user.
+     */
+    public static void farewell() {
```
As farewell is commonly used as a noun and according to standard naming conventions, a method should be named with a verb, should the name of the farewell method be changed to something else that is a verb?

waihin26 committed:
add GUI for chatbot

ruiming97 commented on src/main/java/blue/UI.java:
```diff
@@ -0,0 +1,96 @@
+ * The {@code UI} class handles interactions with the user by displaying messages and task information.
+ * It provides methods to greet the user, say farewell, and display task-related updates.
+ */
+public class UI {
```
Based on standard naming convention, abbreviation (UI) shouldn't be uppercase when used as part of a name. Should it be Ui instead for the name of this class?

ruiming97 commented on src/main/java/blue/Blue.java:
```diff
@@ -0,0 +1,41 @@
+    /**
+     * Runs the Blue application by greeting the user, processing user input, and saying farewell.
+     */
+    public void run() {
```
I like how the name run is used here which obeys the standard naming convention (the name of a method should be a verb)

ruiming97 commented on src/main/java/blue/Storage.java:
```diff
@@ -0,0 +1,58 @@
+public class Storage {
+
+    /** The path to the file where tasks are saved and loaded from. */
+    private static final String FILE_PATH = "./data/blue.txt";
```
Correct naming format for a constant

waihin26 committed:
complete GUI interface

waihin26 committed:
Merge branch 'Level-10'

waihin26 committed:
implement checkstyle

Nihirraa commented on src/main/java/blue/Storage.java:
```diff
@@ -0,0 +1,58 @@
+package blue;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
```
I like how you have followed the importing convention here while importing packages.

Nihirraa commented on src/main/java/blue/task/Task.java:
```diff
@@ -0,0 +1,123 @@
+        boolean isDone = parts[1].equals("1");
+        String description = parts[2];
+
+        Task task;
```
can initialise task with
``` suggested
Task task = null;
```

Nihirraa commented on src/main/java/blue/task/TaskList.java:
```diff
@@ -0,0 +1,182 @@
+        System.out.println(task.toString());
+    }
+
+    /**
+     * Marks the specified task as done.
+     *
+     * @param number The task number to mark as done.
+     * @throws WrongNumberOfItemException If the task number is invalid.
+     */
```
I like the use of comments before every function.

Nihirraa commented on src/main/java/blue/Parser.java:
```diff
@@ -0,0 +1,200 @@
+        }
+    }
+
+    public String parse(TaskList taskList) {
+        String input = "";
+
+        while (true) {
+            input = scanner.nextLine();
+
+            // Exit the loop if "bye" is typed
+            if (input.equalsIgnoreCase("bye")) {
+                break;
+            }
+
+            // Print the task list if "list" is typed
+            if (input.equalsIgnoreCase("list")) {
+                taskList.printList();
+                continue;
+            }
+
+            // Mark a task as done
+            if (input.startsWith("mark ")) {
+                try {
+                    int taskNumber = Integer.parseInt(input.substring(5));
+                    taskList.mark(taskNumber);
+                } catch (NumberFormatException e) {
+                    return "Invalid task number format! Please use 'mark <number>'.";
+                } catch (WrongNumberOfItemException e) {
+                    return "Please check the number you input.";
+                }
+                continue;
+            }
```
I like the use of Egyptian style Braces in the entire code

Nihirraa commented on src/main/java/blue/task/TaskList.java:
```diff
@@ -0,0 +1,182 @@
+    private ArrayList<Task> myList;
+
+    /** The number of tasks in the list. */
+    private int noOfTask;
```
instead of using this name, can use a better more accurate name like:
``` suggested
private int taskCount;
```

Nihirraa reviewed (commented):
Overall, I found the code easy to understand and well written following good coding conventions which made navigation through the code easier.

waihin26 committed:
Add assert statements in Blue class for validating object initialization

The current Blue class does not validate that critical objects like
TaskList and Parser are initialized before being used. This could
lead to potential issues where these objects are unexpectedly null,
causing runtime exceptions.

To prevent such issues, assert statements are added in the constructor
to ensure that both TaskList and Parser are initialised properly.
Additionally, an assert statement is introduced in the getResponse
method to verify that the input is not null before it is processed.

These assertions serve to document important assumptions in the code
and catch potential errors early during development. Using Java’s
built-in assert functionality ensures that these checks are only
evaluated during testing or when explicitly enabled, keeping production
performance unaffected.

Let's use assertions to provide a safeguard for critical assumptions
and to improve code reliability during development.

waihin26 committed:
add Assert checks for Parser and Storage Class

There are several methods that require the arguments to be non-null but
currently do not implement any checks.

Adding a few assert checks ensures that the errors are handled and not thrown to the user.

waihin26 committed:
Improve code quality for Parser Class.

Initial Parser class had one big nested loop. Messy and hard to debug.

Parser class not implements separate methods to deal with different inputs.

waihin26 committed:
override equals method for EventTask class

waihin26 committed:
Add detect duplicate function

waihin26 committed:
Merge branch 'branch-A-Assertions'

waihin26 committed:
no message

waihin26 committed:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	src/main/java/blue/Parser.java

waihin26 committed:
Merge branch 'branch-DetectDuplicates'

waihin26 committed:
add welcome message. Bot now starts with a welcome message.

waihin26 committed:
add user guide documentation

waihin26 committed:
update ReadMe

waihin26 committed:
change Ui image

waihin26 committed:
update readMe

waihin26 committed:
update readme

waihin26 committed:
add link on readMe

waihin26 committed:
add code to readme

waihin26 committed:
README: Add image to improve User UI

The README lacks an image demonstrating the User UI.

Adding an image helps users understand the User UI better and
improves the overall documentation.

Let's update the README by adding an image to showcase the User UI.

This visual aid enhances user comprehension and engagement with
the documentation.

waihin26 committed:
Clarify comment for run() method

The comment for the run() method is not detailed enough to explain its
full functionality.

This can lead to misunderstandings for developers trying to grasp how
the method operates.

Let's update the comment to provide a clearer description of how the
method greets the user, processes input, and manages the application
flow.

Enhancing the comment improves code readability and aids future
maintenance.

waihin26 committed:
Parser.java: Clarify comment for parse() method

The comment for the parse() method is not sufficiently detailed to explain
how user commands are processed and actions are triggered.

This can lead to misunderstandings for developers trying to understand the
command parsing logic.

Let's update the comment to provide a clearer description of how the method
interprets user inputs and interacts with the TaskList.

Enhancing the comment improves code readability and aids future maintenance.

waihin26 committed:
InputErrorException.java: Clarify comment for default constructor

The comment for the default constructor is not sufficiently detailed
to explain the default error message provided.

This may lead to confusion for developers using this exception class.

Let's update the comment to explicitly state the default error message
and its intended use.

Enhancing the comment improves code documentation and aids future
maintenance.

waihin26 committed:
Update README.md

waihin26 committed:
README: Add image to improve User UI

The README lacks an image demonstrating the User UI.

Adding an image helps users understand the User UI better and
improves the overall documentation.

Let's update the README by adding an image to showcase the User UI.

This visual aid enhances user comprehension and engagement with
the documentation.

waihin26 committed:
Merge branch 'master' of https://github.com/waihin26/ip

waihin26 committed:
README: Update comment to improve User UI

The README lacks an image demonstrating the User UI.

Adding an image helps users understand the User UI better and
improves the overall documentation.

Let's update the README by adding an image to showcase the User UI.

This visual aid enhances user comprehension and engagement with
the documentation.

waihin26 committed:
Merge pull request #2 from waihin26/A-MergingPr

README: Update comment to improve User UI
